decompy/RL/Model/Model_template.py

```diff
@@ -38,19 +38,61 @@ def get_past_info(self, summary_of_current_state, possible_actions):
             """
             pass
 
+        def get_decision_by_hash(self, hashed_llvm):
+            """
+            gets the decision by hashed_llvm summary.
+            :param hashed_llvm: the hashed_llvm summary
+            :return: a decision or None if not found.
+            :rtype: Decision
+            """
+            return self.decision_history[hashed_llvm]
+
+        def get_all_decisions_dictionary(self, hashed_llvm):
+            """
+            gets the decision by hashed_llvm summary.
+            :param hashed_llvm: the hashed_llvm summary
+            :return: a decision or None if not found.
+            :rtype: Decision
+            """
+            dict_list = []
+            # loop through all keys, add them to a new list
+            for key, value in self.decision_history:
+                if type(value) is list:
+                    dict_list.extend(value)
+                else:
+                    dict_list.append(value)
+            return self.decision_history[hashed_llvm]
+
+        def get_all_decisions_list(self):
+            """
+            gets all decisions and puts them into a list.
+            :return: a decision list
+            :rtype: list<Decision>
+            """
+            return (for _ in self.decision_history.iteritems())
+
         def update_model(self, decision):
             """
             Updates the model based off a decision.
             :param decision: the decision to update the model
             :type: Decision
             :return: nothing
             """
-            pass
+            # get hash representation from llvm summary
+            hashed = hash(decision.summary)
+
+            # if it already exists, then add it to a list
+            if hashed in self.decision_history:
+                decision_list = self.decision_history[hashed]
+                decision_list.append(decision)
+                decision = decision_list  # for convenience
+
+            self.decision_history[hashed] = decision
 
     def __init__(self, decision_history):
```

Comment:
I feel like it would make more sense for "decision_history" to default to an empty list. I don't think the main use case is to put in a decision history seeing as there should only be one model in this program and when the model doesn't exist yet we won't have a list of decisions.